File: leetcode-swift/lc_026.swift
//
// LeetCode
// 26. Remove Duplicates from Sorted Array
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
//


class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        if nums.count <= 1 {
            return nums.count
        }

        var lastIndex = 0
        for num in nums {
            if num != nums[lastIndex] {
                lastIndex += 1
                nums[lastIndex] = num
            }
        }

        return lastIndex + 1
    }
}

File: leetcode-python/lc_026.py
#
# LeetCode
# 26. Remove Duplicates from Sorted Array
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/
#

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0: return 0

        write_index = 0
        for i in range(len(nums)):
            if nums[write_index] != nums[i]:
                write_index += 1
                nums[write_index] = nums[i]

        return write_index + 1
